mx-kinetic-scroll-view: Add a new scroll policy to align scrolling based on initial interaction

ManMower opened the pull request:
(I accidentally mixed this in with a different pull request that has since been merged, so I'm recreating this one on its own)

This patch adds a SCROLL_POLICY_ALIGNED for which:
If a scroll is started in a mostly vertical direction the view will only scroll vertically.
If a scroll is started in a mostly horizontal direction the view will only scroll horizontally.
If a scroll is started diagonally the view will scroll in both directions.

While recent changes to MX allow the first two to be achieved by nesting scrolling widgets, there is not yet a way to have such a nested configuration scroll in both directions if the initial vector is "Mostly diagonal"

thos commented:
This looks like a good idea, but could we name the policy "automatic", rather than "aligned"? I think this better explains that the policy is automatically decided, based on the initial direction of the scroll.

Commit:
Add a new aligned scroll policy to mx-kinetic-scroll-view

If a scroll is started in a mostly vertical direction the view will only scroll vertically.
If a scroll is started in a mostly horizontal direction the view will only scroll horizontally.
If a scroll is started diagonally the view will scroll in both directions.

ManMower commented:
Ok, I've renamed aligned to automatic

thos commented:
Thanks for the patch, it has been committed to master.
